### PR TITLE
Switch from Travis to Circle for CI testing

### PR DESCRIPTION
**Pull request by graysonwright**
- Copy .sample.env to .env before testing to ensure environment
  variables are set correctly
- Remove travis configuration file

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,24 +0,0 @@
...
-  - cp .sample.env .env.test
-
-services:
-  - redis-server
-
-notifications:
-  slack:
-    secure: BdIl+dAUtA6gLB9lk/4xNYoERF+yIcFp8J+wNAzMV4eNjEOC6fAaZyFCxCjv2CjDT1lWxWLPzfSyoCDjxI7A5ykQLlcI25d7Pk0SjclOq47BqIyZRSKt1ZMHqh928YfRZZKx4AUibXsx5draI74+5btU11ocV3wVKIB9yAXhvig=
-    on_success: change
-    on_failure: always
```
Comment (croaky) [inline .travis.yml]: Can we get Slack notifications?

Comment (graysonwright) [inline .travis.yml]: were those just for the master branch?

**Comment by croaky** [comment]
Looks like REDIS_URL is necessary https://circleci.com/gh/thoughtbot/hound/23

File: .travis.yml (hunk 1 above)
Comment (thorncp) [inline .travis.yml]: I remember master failing to bubbles a handful of times, but don't recall a slack notification for any of them. Only master failures makes sense to me though.

Comment (graysonwright) [inline .travis.yml]: Looks like it will post notifications for all builds, not just master.
https://thoughtbot.slack.com/services/6337427525?updated=1

**Comment by graysonwright** [comment]
Added a pre-build script to copy `.sample.env` to `.env` - that gives us a default `REDIS_URL`, along with all the other required environment variables.

It also forces us to keep `.sample.env` up to date. :smile:

**Comment by croaky** [comment]
Passing build!

**Comment by gylaz** [comment]
So much red! :+1:

**Comment by graysonwright** [comment]
woohoo! Travis is still thinking about it...

**Comment by croaky** [comment]
How about parallelizing?

https://circleci.com/docs/parallel-manual-setup

**Comment by graysonwright** [comment]
That's a good idea - it would probably speed up the tests a bit.

I think I'll hold off for now, because a lot of other people have failing PRs that depend on this one. I'll open another PR for parallelization.

**Pull request merged**